From 4a6353dad68074bb7ae08817cd7beb5cd13f2359 Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Mon, 18 Jun 2012 12:02:56 -0300
Subject: [PATCH] NFS: Fix nfsroot support

The NFS support to use alternative ports breaks building of regular
NFS root support as the API has been change in an incompatible way.

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
---
 fs/nfs/nfsroot.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/nfs/nfsroot.c b/fs/nfs/nfsroot.c
index df101d9..e1578e1 100644
--- a/fs/nfs/nfsroot.c
+++ b/fs/nfs/nfsroot.c
@@ -506,7 +506,7 @@ static int __init root_nfs_get_handle(void)
 	if (!request.fh)
 		goto out;
 	set_sockaddr(&sin, servaddr, htons(mount_port));
-	status = nfs_mount(&request);
+	status = nfs_mount(&request, NFS_MNT_PROGRAM);
 	if (status < 0)
 		printk(KERN_ERR "Root-NFS: Server returned error %d "
 				"while mounting %s\n", status, nfs_export_path);
-- 
1.7.10

